great_expectations/expectations/expectation.py

```diff
@@ -347,6 +347,11 @@ def __eq__(self, other: object) -> bool:
         self_dict = self.dict()
         other_dict = other.dict()
 
+        # rendered_content is derived from the rest of the expectation, and can/should
+        # be excluded from equality checks
+        self_dict.pop("rendered_content", None)
+        other_dict.pop("rendered_content", None)
```

Comment:
```suggestion
        # rendered_content is derived from the rest of the expectation, and can/should
        # be excluded from equality checks
        exclude: set[str] = {"rendered"}
       
        self_dict = self.dict(exclude=exclude)
        other_dict = other.dict(exclude=exclude)
        self_dict = self.dict()
        other_dict = other.dict()
```

Reply:
I personally like just colocating it with where its used, since as of now we don't have a use case for using it outside of this method.

Reply:
I don't feel strongly.